File: python/bubble_sort.py
def bubble_sort(arr: list) -> list:
    """
    冒泡排序
    @param arr: 待排序数组
    @return: 排序后的数组
    """
    n =len(arr)
    for i in range(n):
        swapped = False
        for j in range(n-1-i):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = True
        if not swapped:
            break
    return arr

File: rust/src/bubble_sort.rs
pub fn bubble_sort<T: PartialOrd + Copy>(arr: &mut [T]) {
    let n = arr.len();
    for i in 0..n {
        let mut swapped = false;
        for j in 0..n - 1 - i {
            if arr[j] > arr[j + 1] {
                arr.swap(j, j + 1);
                swapped = true;
            }
        }
        if !swapped {
            break;
        }
    }
}
